### DISCUSSION TITLE
Next.js Example of withUrqlClient with Per Page Layout

### DISCUSSION THREAD

**Discussion by mtergel** [urql-graphql/urql#2064] [Q&A] [answered]
Need help implementing withUrqlClient with [Per Page Layout](https://nextjs.org/docs/basic-features/layouts#per-page-layouts)

Where or how to add getLayout function?
```
export default withUrqlClient(
  (ssr) => ({
    url: "###",
  }),
  { ssr: false } // Important so we don't wrap our component in getInitialProps
)(Profile);
```

**Comment by JoviDeCroock** [comment, marked as answer]
Should work by doing

```
const withUrql = withUrqlClient(
  (ssr) => ({
    url: "###",
  }),
  { ssr: false } // Important so we don't wrap our component in getInitialProps
)(Profile);

withUrql. getLayout

export default withUrql
```

But I think even `Profile. getLayout` should work if we implement that feature (similar to `getInitialProps`)

This should cover that in the future https://github.com/FormidableLabs/urql/pull/2065

**Reply by boompttn** [reply]
What should we do if we're using Typescript and using a different name other than `getLayout`? 
<img width="574" alt="image" src="https://user-images.githubusercontent.com/37561295/193181533-4b2073b5-04f0-4b0a-9671-4b644f58924d.png">
